packages/rum-react/test/specs/get-with-transaction.spec.js

```diff
@@ -79,16 +84,33 @@ describe('withTransaction', function() {
   })
 
   it('should return WrappedComponent on falsy value and log warning', function() {
-    const serviceFactory = createServiceFactory()
     const loggingService = serviceFactory.getService('LoggingService')
-
     spyOn(loggingService, 'warn')
 
-    const withTransaction = getWithTransaction(new ApmBase(serviceFactory))
+    const withTransaction = getWithTransaction(apmBase)
     const comp = withTransaction('test-name', 'test-type')(undefined)
     expect(comp).toBe(undefined)
     expect(loggingService.warn).toHaveBeenCalledWith(
       'test-name is not instrumented since component property is not provided'
     )
   })
+
+  it('should not instrument the route when rum is inactive', () => {
+    const transactionService = serviceFactory.getService('TransactionService')
+    spyOn(transactionService, 'startTransaction')
+
+    apmBase.config({ active: false })
+    TestComponent(apmBase)
+
+    function Component() {
+      return <h2>Component</h2>
+    }
+    const withTransaction = getWithTransaction(apmBase)
+
+    const WrappedComponent = withTransaction('test-transaction', 'test-type')(
+      Component
+    )
+    expect(WrappedComponent).toEqual(Component)
+    expect(transactionService.startTransaction).not.toHaveBeenCalled()
```

Comment:
`transactionService.startTransaction` won't be called unless the component is rendered, so this probably always passes regardless of the test!

Reply:
Yes I added the component to get rendered as well in previous line for double checking. 
https://github.com/elastic/apm-agent-rum-js/blob/046b6400a835f539f79fd60ce8a84a77688fd4bc/packages/rum-react/test/specs/get-with-transaction.spec.js#L103

if you comment out the config.isActive check in the source, this test would fail which is expected.

Reply:
Ok, I thought maybe you intended to test with the component that you created in the test instead instead of the one created in `TestComponent`. Might be a bit better to group the code together so it's clear which checks are intended for those parts.